### PR TITLE
Add hint for broadcast join.

### PR DESCRIPTION
**Pull request by hanfei1991**
### What problem does this PR solve? 
Add hint for broadcast join.

### What is changed and how it works?
Let parser support two new hints:
- BC_JOIN(...)
- TIDB_BCJ(...)

`BC` meas broadcast, which is brief and easy to understand :)

### Check List 

Tests 

 - Unit test

Code changes

only change parser

Side effects

on side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

File: ast/misc.go [hunk 1]
```diff
@@ -2539,7 +2539,7 @@ func (n *TableOptimizerHint) Restore(ctx *format.RestoreCtx) error {
 	switch n.HintName.L {
 	case "max_execution_time":
 		ctx.WritePlainf("%d", n.HintData.(uint64))
-	case "tidb_hj", "tidb_smj", "tidb_inlj", "hash_join", "merge_join", "inl_join", "bc_join", "tidb_bcj":
+	case "tidb_hj", "tidb_smj", "tidb_inlj", "hash_join", "merge_join", "inl_join", "bc_join", "bcj_local", "tidb_bcj":
```
Comment (zz-jason) [inline ast/misc.go:2542]: Nit: according to https://www.abbreviations.com/abbreviation/BROADCAST, seems `BRDCST` is a proper abbreviation for `broadcast`

Comment (hanfei1991) [reply ast/misc.go:2542]: It seems brdcst is for media, I think it's quite verbose. Can we make it brief and easy to write ?

Comment (kennytm) [reply ast/misc.go:2542]: if you shorten it to "brdcst" i'd rather you just spell it out entirely

```sql
select /*+ broadcast_join(...) */ ...
```

Comment (zz-jason) [reply ast/misc.go:2600]: @hanfei1991 @lzmhhh123 PTAL the comment from @kennytm

Comment (lzmhhh123) [reply ast/misc.go:2600]: I support the `bc_join` as the abbreviation.

**Comment by hanfei1991** [comment]
@zz-jason PTAL

**Comment by kennytm** [comment]
just wanna note that both `bc_join` and `broadcast_join` are fine for me, but `brdcst_join` is bad.

**Pull request merged**